nifi-nar-bundles/nifi-social-media-bundle/nifi-twitter-processors/src/main/java/org/apache/nifi/processors/twitter/ConsumeTwitter.java

```diff
@@ -0,0 +1,366 @@
+            .name("success")
+            .description("FlowFiles containing an array of one or more Tweets")
+            .build();
+
+    private List<PropertyDescriptor> descriptors;
+    private Set<Relationship> relationships;
+
+    private TweetStreamService tweetStreamService;
+
+    private volatile BlockingQueue<String> messageQueue;
+
+    @Override
+    protected void init(ProcessorInitializationContext context) {
+        final List<PropertyDescriptor> descriptors = new ArrayList<>();
+        descriptors.add(ENDPOINT);
+        descriptors.add(BASE_PATH);
+        descriptors.add(BEARER_TOKEN);
+        descriptors.add(QUEUE_SIZE);
+        descriptors.add(BATCH_SIZE);
+        descriptors.add(BACKOFF_ATTEMPTS);
+        descriptors.add(BACKOFF_TIME);
+        descriptors.add(MAXIMUM_BACKOFF_TIME);
+        descriptors.add(CONNECT_TIMEOUT);
+        descriptors.add(READ_TIMEOUT);
+        descriptors.add(BACKFILL_MINUTES);
+        descriptors.add(TWEET_FIELDS);
+        descriptors.add(USER_FIELDS);
+        descriptors.add(MEDIA_FIELDS);
+        descriptors.add(POLL_FIELDS);
+        descriptors.add(PLACE_FIELDS);
+        descriptors.add(EXPANSIONS);
+
+        this.descriptors = Collections.unmodifiableList(descriptors);
+
+        final Set<Relationship> relationships = new HashSet<>();
+        relationships.add(REL_SUCCESS);
+        this.relationships = Collections.unmodifiableSet(relationships);
+    }
+
+    @Override
+    public Set<Relationship> getRelationships() {
+        return this.relationships;
+    }
+
+    @Override
+    public final List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        return this.descriptors;
+    }
+
+
+    @OnScheduled
+    public void onScheduled(final ProcessContext context) {
+        messageQueue = new LinkedBlockingQueue<>(context.getProperty(QUEUE_SIZE).asInteger());
+
+        tweetStreamService = new TweetStreamService(context, messageQueue, getLogger());
+        tweetStreamService.setBasePath(context.getProperty(BASE_PATH).getValue());
+        final String endpointName = context.getProperty(ENDPOINT).getValue();
+        if (ENDPOINT_SAMPLE.getValue().equals(endpointName)) {
+            tweetStreamService.start(StreamEndpoint.SAMPLE_ENDPOINT);
+        } else {
+            tweetStreamService.start(StreamEndpoint.SEARCH_ENDPOINT);
+        }
+    }
+
+    @Override
+    public void onTrigger(ProcessContext context, ProcessSession session) throws ProcessException {
+        if (messageQueue.isEmpty()) {
+            context.yield();
+            return;
+        }
+
+        FlowFile flowFile = session.create();
+        flowFile = session.write(flowFile, new OutputStreamCallback() {
+            @Override
+            public void process(OutputStream out) throws IOException {
+
+                final int batchSize = context.getProperty(BATCH_SIZE).asInteger();
+                String tweet = messageQueue.poll();
+                int tweetCount = 1;
```

Comment:
The `tweetCount` variable is not used. It is useful information, so recommend setting a FlowFile attribute named `tweets` with the number. That may involve the need to create an AtomicInteger outside of the callback, which can be updated and referenced after the callback completes.

Reply:
Making the changes